File: app/src/main/java/dev/yiray/qickmvivm/ui/home/HomeView.java
package dev.yiray.qickmvivm.ui.home;

import dev.yiray.qickmvivm.model.Task;
import io.reactivex.rxjava3.core.Observable;

public interface HomeView {
    interface Action {
        Observable<CharSequence> observableInput();
        Observable<Boolean> observableCheckBox();
        Observable<Boolean> observableNextPage();
        Observable<UpdateTask> observableUpdated();
    }

    final class UpdateTask {
        private int position;
        private Task task;

        public UpdateTask(int position, Task task) {
            this.position = position;
            this.task = task;
        }

        public int getPosition() {
            return position;
        }

        public Task getTask() {
            return task;
        }
    }
}

File: app/src/main/java/dev/yiray/livedatamvi/ui/home/HomeView.kt
package dev.yiray.livedatamvi.ui.home

import dev.yiray.livedatamvi.model.Task
import kotlinx.coroutines.flow.Flow

interface HomeView {
    interface Action {
        fun observableInput(): Flow<CharSequence>
        fun observableCheckBox(): Flow<Boolean>
        fun observableNextPage(): Flow<Boolean>
        fun observableUpdated(): Flow<UpdateTask>?
    }

    class UpdateTask(val position: Int, val task: Task)
}